src/Mobs/Monster.cpp

```diff
@@ -1158,11 +1158,11 @@ void cMonster::AddRandomUncommonDropItem(cItems & a_Drops, float a_Chance, short
 void cMonster::AddRandomRareDropItem(cItems & a_Drops, cItems & a_Items, unsigned int a_LootingLevel)
 {
 	MTRand r1;
-	unsigned int Count = r1.randInt() % 200;
+	unsigned int Count = static_cast<unsigned int>(static_cast<unsigned long>(r1.randInt()) % 200);
 	if (Count < (5 + a_LootingLevel))
 	{
-		int Rare = r1.randInt() % a_Items.Size();
-		a_Drops.push_back(a_Items.at(Rare));
+		int Rare = static_cast<int>(static_cast<unsigned long>(r1.randInt()) % a_Items.Size());
```

Comment:
I think `unsigned long` should be a `size_t` instead, to match the `.Size()` return type.